src/tests/test_split.cpp

```diff
@@ -0,0 +1,141 @@
+#include <list>
+#include <set>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "all_csv_configs.h"
+#include "config/names.h"
+#include "csv_config_util.h"
+#include "types.h"
+
+namespace tests {
+
+void CompareDDStringLists(std::set<std::pair<std::set<model::DFStringConstraint>,
+                                             std::set<model::DFStringConstraint>>> const& expected,
+                          std::list<model::DDString> const& actual) {
+    ASSERT_EQ(expected.size(), actual.size())
+            << "count of generated dependencies does not match: expected " << expected.size()
+            << ", got " << actual.size();
+    for (auto const& dd : actual) {
+        std::set<model::DFStringConstraint> lhs(dd.left.begin(), dd.left.end());
+        std::set<model::DFStringConstraint> rhs(dd.right.begin(), dd.right.end());
+        if (expected.find(std::make_pair(std::move(lhs), std::move(rhs))) == expected.end()) {
+            FAIL() << "generated dependency that is not expected";
+        }
+    }
+}
+
+class SplitAlgorithmTest : public ::testing::Test {
+public:
+    static algos::StdParamsMap GetParamMap(CSVConfig const& csv_config,
+                                           CSVConfig const& dif_table_csv_config) {
+        using namespace config::names;
+        if (dif_table_csv_config.path == "") {
+            return {{kCsvConfig, csv_config}};
+        }
+        return {{kCsvConfig, csv_config}, {kDifferenceTable, MakeInputTable(dif_table_csv_config)}};
+    }
+
+    static std::unique_ptr<algos::dd::Split> CreateSplitAlgorithmInstance(
+            CSVConfig const& csv_config, CSVConfig const& dif_table_csv_config = {}) {
+        return algos::CreateAndLoadAlgorithm<algos::dd::Split>(
+                GetParamMap(csv_config, dif_table_csv_config));
```

Comment:
it looks like we should use `std::optional<CSVConfig>` for dif table (in both methods)